File: DSA/Assignment_4/del.c
#include <stdio.h>
#include <stdlib.h>
typedef struct Node
{
    int data;
    struct Node *next;
}Node;
int main()
{
    int n;
    printf("Enter number of nodes:");
    scanf("%d",&n);
    Node *head=NULL;
    Node *tail=NULL;
    for(int i=0;i<n;i++)
    {
        Node *cur=(Node *)malloc(sizeof(Node));
        printf("Enter element:");
        scanf("%d",&cur->data);
        cur->next=NULL;

        if(head==NULL)
        {
            head=cur;
            tail=cur;
        }
        else
        {
            tail->next=cur;
            tail=cur;
        }
    }

    printf("original Linked List:");
    Node *temp=head;
    while(temp)
    {
        prnitf("%d ",temp->data);
        temp=temp->next;
    }
    printf("\n");

    int del_pos;
    printf("Enter delete position:");
    scanf("%d",&del_pos);
    temp=head;
    while(temp && del_pos>0)
    {
        temp=temp->next;
        del_pos--;
    }
    
    
    return 0;
}

File: DSA/Assignment_4/del.py
class Node:
    def __init__(self, data):
        self.data=data
        self.next=None

class Linked_list:
    def __init__(self):
        self.head=None

    def insert_ele(self, data):
        new_node=Node(data)
        if self.head is None:
            self.head=new_node
            return
        cur=self.head
        while cur.next:
            cur=cur.next
        cur.next=new_node

    def print_list(self):
        temp=self.head
        while temp:
            print(temp.data,end=' ')
            temp=temp.next
        print()

    def del_list(self, pos):
        temp=self.head
        if pos==0:
            self.head=temp.next
            temp=None
            return
        p=1
        while temp and p<pos:
            prev=temp
            temp=temp.next
            p+=1
        prev.next=temp.next
        temp=None


list=Linked_list()

n=int(input("Enter number of elements:"))
for i in range(n):
    ele=int(input("Enter element:"))
    list.insert_ele(ele)

print("Original List:",end=' ')
list.print_list()

del_pos=int(input("Enter position of element to delete:"))
list.del_list(del_pos)

print("modified List:")
list.print_list()
